### PR TITLE
Master

### PR DESCRIPTION
**Pull request by clozr**
Added rnpm link support

### CODE REVIEW

File: rnpm-postlink.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env node
+
+'use strict';
+
+var fs = require('fs');
+var rnpm = require('rnpm/src/config');
+var xcode = require('xcode');
+
+// The current working directory should be project root of the app that is linking Realm.
```
Comment (andpor) [inline rnpm-postlink.js]: Can you remove the references to Realm here and replace with react-native-storage?

File: rnpm-postlink.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    ['libsqlite3.0'].forEach(function(name) {
+        project.addFramework('usr/lib/' + name + '.tbd', {
+            lastKnownFileType: 'sourcecode.text-based-dylib-definition',
+            sourceTree: 'SDKROOT',
+            target: target,
+        });
+    });
+
+    fs.writeFileSync(pbxproj, project.writeSync());
+}
```
Comment (andpor) [inline rnpm-postlink.js]: This is only once step. Would it be possible to add step to check if react-native-storage is linked as a sub-project as well and link it up if it is not?

File: README.md [hunk 3]
```diff
@@ -51,27 +51,16 @@ v1.0 - Intial release for iOS with full support of all operations based on plan
...
 
-#### Step 2. XCode SQLite project dependency set up
-
-Drag the SQLite Xcode project as a dependency project into your React Native XCode project
-
-![alt tag](https://raw.github.com/andpor/react-native-sqlite-storage/master/instructions/libs.png)
-
-#### Step 3. XCode SQLite libraries dependency set up
-
-Add libSQLite.a (from Workspace location) to the required Libraries and Frameworks. Also add sqlite3.0.tbd (XCode 7) or libsqlite3.0.dylib (XCode 6 and earlier) in the same fashion using Required Libraries view (Do not just add them manually as the build paths will not be properly set)
```
Comment (andpor) [inline README.md]: Are you sure that your script is actually handling the project linkage and  libSQLite.a steps? I think I recognized the Frameworks add for sqlite library only?

File: package.json [hunk 4]
```diff
@@ -31,5 +31,17 @@
...
   "peerDependencies": {
     "react-native": ">=0.14.0"
+  },
+  "dependencies": {
+    "rnpm": "1.5.2",
+    "xcode": "0.8.4"
+  },
+  "rnpm": {
+    "commands": {
+      "postlink": "node_modules/react-native-sqlite-storage/rnpm-postlink.js"
```
Comment (andpor) [inline package.json]: Can you create a scripts directory and place rnpm-postlink.js there?

File: README.md [hunk 5]
```diff
@@ -51,27 +51,16 @@ v1.0 - Intial release for iOS with full support of all operations based on plan
 
 ```shell
 npm install --save react-native-sqlite-storage
+rnpm link
```
Comment (andpor) [inline README.md]: Is rnpm something that comes bundled with react-native-cli or needs to be downloaded and installed separately? If separately, we need some instructions on how to get it installed before we tell users to run the command.

**Comment by andpor** [comment]
@clozr  - can you attach a screen shot of the xcode project view BEFORE and AFTER you execute rnpm link? Thanks.

**Comment by andpor** [comment]
@clozr - I test your PR but do not see it doing anything. It just runs but the host RN project is not changed at all...

File: package.json [hunk 6]
```diff
@@ -31,5 +31,17 @@
...
+    },
+    "ios": {
+      "project": "react-native-sqlite-storage/src/ios/SQLite.xcodeproj"
```
Comment (andpor) [inline package.json]: this path does not seem to work, src/ios/SQLite.xcodeproj works better? Can you provide me with steps I can use to test your PR ?

File: package.json (hunk 4 above)
Comment (clozr) [inline package.json]: sure I will move it.

File: README.md (hunk 5 above)
Comment (clozr) [inline README.md]: I think rnpm needs to be separately installed.

**Comment by clozr** [comment]
I messed up with the path during check-in. Now it is working alright. I have put the screenshot there as well and moved rnpm-postlink to scripts directory

**Comment by andpor** [comment]
So which version of node and rnpm are you testing this against. Asking because I am not seeing the Frameworks group (and child library) in the project after running the link...i have node 4.1.5 and rnpm 1.6.2

**Comment by andpor** [comment]
@clozr  - I got the link to work finally. Had to npm link xcode locally for some reason. Perhaps it makes sense to leave the original instructions in place as a fallback in case the rnpm link does not work for someone?

**Comment by clozr** [comment]
I updated the rnpm installation instruction to include xcode as well.
We can certainly leave the manual installation instructions.

**Comment by laurent22** [comment]
I don't fully understand what this change does but it's causing some problem on Windows (even thought the package used to work fine on this OS). The error is:

````
> react-native link react-native-sqlite-storage
Scanning folders for symlinks in D:\Web\www\joplin\ReactNativeClient\node_modules (21ms)
rnpm-install info iOS module react-native-sqlite-storage is already linked
rnpm-install ERR! Something went wrong while linking. Error: spawn UNKNOWN
Please file an issue here: https://github.com/facebook/react-native/issues

spawn UNKNOWN
````

Unfortunately the link command doesn't print the full stacktrace but I guess it happens somewhere in:

    ...\node_modules\react-native-sqlite-storage\scripts\rnpm-postlink.js

If I remove the `postlink` command from the `package.json` then linking works again.

**Pull request merged**